custom_components/rainforest_emu_2/emu2.py

```diff
@@ -62,15 +65,23 @@ async def close(self) -> None:
     async def open(self) -> bool:
         if self._connected == True:
             return True
-
-        try:
-            self._reader, self._writer = await serial_asyncio.open_serial_connection(
-                url = self._device,
-                baudrate = 115200
-            )
-        except SerialException as ex:
-            _LOGGER.error(ex)
-            return False
+        if self._host:
+            try:
+                    self._reader, self._writer = await asyncio.open_connection(
+                        self._host, self._port
+                    )
+            except Exception as ex:
+                _LOGGER.error(ex)
+                return False           
+        else:
+            try:
+                    self._reader, self._writer = await serial_asyncio.open_serial_connection(
+                        url = self._device,
+                        baudrate = 115200
+                    )
+            except SerialException as ex:
```

Comment:
I believe this can be simplified to an Exception type, which means the code could use a common exception block and the if self._host could be within the try.